multibody/kinematic/world_point_evaluator.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+#include "multibody/kinematic/kinematic_evaluator.h"
+
+#include "drake/multibody/plant/multibody_plant.h"
+#include "drake/systems/framework/context.h"
+
+namespace dairlib {
+namespace multibody {
+
+/// Basic contact evaluator for a point on a body w.r.t. the world
+template <typename T>
+class WorldPointEvaluator :  public KinematicEvaluator<T> {
+ public:
+  /// The basic constructor for WorldPointEvaluator, defined via a rotation
+  /// matrix.
+  /// @param plant
+  /// @param pt_A the contact point on the body
+  /// @param frame_A the frame of reference for pt_A
+  /// @param rotation The rotation matrix R s.t. the evaluation is R * r_A_world
+  ////   (default identity matrix)
+  /// @param offset The nominal world location of pt_A (default [0;0;0])
+  /// @param active_directions The directions, a subset of {0,1,2} to be
+  ///    considered active. These will correspond to rows of the rotation matrix
+  ////   Default is {0,1,2}.
+
+  WorldPointEvaluator(
+      const drake::multibody::MultibodyPlant<T>& plant,
+      const Eigen::Vector3d pt_A, const drake::multibody::Frame<T>& frame_A,
+      const Eigen::Matrix3d rotation = Eigen::Matrix3d::Identity(),
+      const Eigen::Vector3d offset = Eigen::Vector3d::Zero(),
+      std::vector<int> active_directions_ = {0,1,2});
+
+  /// Constructor for WorldPointEvaluator, defined via a normal direction
+  /// This will automatically construct an appropriate rotation matrix.
+  /// This constructor is primarily used for world contacts, where the normal
+  /// can represent the contact normal, and the tangential directions are
+  /// defined only w.r.t. the normal.
+  /// @param plant
+  /// @param pt_A the contact point on the body
+  /// @param frame_A the frame of reference for pt_A
+  /// @param normal The normal unit vector (must be specified)
+  /// @param offset The nominal world location of pt_A
+  /// @param xy_active If true, then the tangential directions are active
+  ///    If false, they be inactive, and thus still appear in
+  ///    the "Full" terms, but not the "Active" values
+  ///    Note that this defaults to FALSE, while the base constructor above
+  ///    includes all dimensions by default.
+
+  WorldPointEvaluator(
+      const drake::multibody::MultibodyPlant<T>& plant,
+      const Eigen::Vector3d pt_A, const drake::multibody::Frame<T>& frame_A,
+      const Eigen::Vector3d normal,
+      const Eigen::Vector3d offset = Eigen::Vector3d::Zero(),
+      bool tangent_active = false);
```

Comment:
I think it's probably better to use `active_directions` instead of `tangent_active` here so it's more consistent? 
The different between the two constructors in this class should be that one takes `Matrix3d rotation` and the other takes `Vector3d normal`. (I don't see why the second constructor disable the xy direction by default)

Reply:
Ok.